gomock/call.go

```diff
@@ -75,6 +75,9 @@ func (c *Call) Do(f interface{}) *Call {
 	return c
 }
 
+// Return sets the values to return when the target method is called.
+//
+// If Return and Do are called, the values returned via Do will be returned.
```

Comment:
@kaedys Thanks for the analysis. This PR is really an attempt to open the conversation about using this package as a tool to create stub functions.

The main problem here is that after using this mock library for simple cases, one may have to switch over to stubs if it is required to respond to input.

It is also slightly obnoxious that it is not clear that `Do` cannot return a value and it took time to identify this as the problem. A panic for mismatched function signature would be a gift, as it would have exposed the error in my code. Either that, or we have some compile time type-safety to give us the correct stub types.

> On a behavioral side, an explicit call to Return should probably override the return-what-Do-returned behavior (if you're explicitly calling Return, it's because you want the function to return something specific), and the code in general should probably be updated to only return the Dofunc return values if they match the number and type of the parent function itself.

I considered this. Either behavior is problematic, as it may result in test failures for existing uses, making the change backwards compatible.

Perhaps, we should define a new method, `Stub` for this functionality.